airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1622,6 +1693,31 @@ def create_declarative_stream(
                 config=config,
                 parameters=model.parameters or {},
             )
+        elif model.incremental_sync and isinstance(
+            model.incremental_sync, IncrementingCountCursorModel
+        ):
+            cursor_model: IncrementingCountCursorModel = model.incremental_sync  # type: ignore
+
+            start_time_option = (
+                self._create_component_from_model(
+                    cursor_model.start_value_option,  # type: ignore # mypy still thinks cursor_model of type DatetimeBasedCursor
+                    config,
+                    parameters=cursor_model.parameters or {},
+                )
+                if cursor_model.start_value_option  # type: ignore # mypy still thinks cursor_model of type DatetimeBasedCursor
+                else None
+            )
+
+            # The concurrent engine defaults the start/end fields on the slice to "start" and "end", but
+            # the default DatetimeBasedRequestOptionsProvider() sets them to start_time/end_time
+            partition_field_start = "start"
+
+            request_options_provider = DatetimeBasedRequestOptionsProvider(
```

Comment:
The naming here is weird. Should we generify the `DatetimeBasedRequestOptionsProvider` to be a `StartEndRequestOptionsProvider` or something like that? Would this allow us to have only one method for building both `DatetimeBasedCursor` and `IncrementingCountCursor`? The comments above refer to `DatetimeBasedCursor` which makes me think there is duplication